radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java

```diff
@@ -215,6 +283,55 @@ private byte[] getBytes(Particle particle) {
   }
 
   private SubstateSerialization getSubstateSerializer() {
-    return this.forks.get(epoch).serialization();
+    return this.forks.get(lastEpochInDbOpt.orElse(0L)).serialization();
+  }
+
+  /* When we are updating the epoch hash file (appending new epoch hashes), we must not have processed epochs greater
+  than the last one in the epoch hash file. Otherwise, there is no way of appending new epochs without having gaps. */
+  private void assertEpochInTheDbIsNotGreaterThanEpochInTheFile(Optional<Long> lastEpochInFileOpt)
+      throws IllegalStateException {
+    boolean isValid = true;
+    if (lastEpochInFileOpt.isEmpty()) {
+      if (this.lastEpochInDbOpt.isPresent()) {
+        // there is an epoch in db, but none in the file
+        isValid = false;
+      }
+    } else if (this.lastEpochInDbOpt.map(it -> it > lastEpochInFileOpt.get()).orElse(false)) {
+      // we have an epoch in the file and need to check it is not greater than the one in the db (if
+      // any)
+      isValid = false;
+    }
+    if (!isValid) {
+      throw new IllegalStateException(
+          String.format(
+              "Epoch in db %s is greater than epoch in file %s",
```

Comment:
If someone's node crashes with this exception, what should they do? Can you explain in the error message the likely cause/fix?

Reply:
I believe the only fix is to clean the ledger. Is it good "Please delete the content in the db folder and try again."?